File: c#/Temporizador.cs
using System;
using System.Diagnostics;

namespace queries{

    class Temporizador{
        public static Stopwatch watch;
        public static int hilosFinalizados;

        public void IniciarTemporizador(){
            watch = Stopwatch.StartNew();
        }

        public void finalizarHilo(){
            hilosFinalizados++;
            if (hilosFinalizados == 10){
                finalizarTemporizador();
            }

        }

        public void finalizarTemporizador(){
            watch.Stop();
            long elapsedMs = watch.ElapsedMilliseconds;
            Console.WriteLine("Tomó " + elapsedMs + " milisegundos");
        }
    }
}

File: java/Temporizador.java
public class Temporizador {
    public static int hilosFinalizados = 0;
    public static long inicio;

    public void iniciarTemporizador(){
        inicio = System.currentTimeMillis();
    }

    public void finalizarHilo(){
        hilosFinalizados++;
        if(hilosFinalizados==10){
            finalizarTemporizador();
        }
    }

    public void finalizarTemporizador(){
        long tiempo = System.currentTimeMillis() - inicio;
        System.out.println("Tomó " + tiempo + " milisegundos.");
    }
}
